src/dashboard/tests/test_ingest.py

```diff
@@ -76,3 +77,187 @@ def test_add_metadata_files_view(self):
+        "format": None,
+    }
+    mapping = {}
+    # this parameter is modified in place and will contain
+    # formatted nodes for each transfer
+    nodes = []
+    _es_results_to_appraisal_tab_format(record, mapping, nodes)
+    # extract the only transfer node at the moment
+    transfer_node = nodes[0]
+    assert transfer_node["type"] == "transfer"
+    # there are child nodes for each directory in the path
+    data_dir_node = transfer_node["children"][0]
+    assert data_dir_node["type"] == "directory"
+    objects_dir_node = data_dir_node["children"][0]
+    assert objects_dir_node["type"] == "directory"
+    # and finally a node for the file
+    file_node = objects_dir_node["children"][0]
+    # check that the file node properties match the record properties
+    _assert_file_node_properties_match_record(file_node, record)
+
+
+def test_appraisal_tab_node_formatter_old_transfer_format():
+    """
+    Test the _es_results_to_appraisal_tab_format helper works with the
+    pre bag transfer layout (no data directory).
+    """
+    record = {
+        "relative_path": "transfer-directory/objects/MARBLES.TGA",
+        "fileuuid": "ea7a98d4-dcb5-46b3-a9dd-82262bd983aa",
+        "size": 4.0,
+        "tags": ["tag1", "tag2"],
+        "bulk_extractor_reports": [],
+        "modification_date": None,
+        "format": None,
+    }
+    mapping = {}
+    # this parameter is modified in place and will contain
+    # formatted nodes for each transfer
+    nodes = []
+    _es_results_to_appraisal_tab_format(record, mapping, nodes)
+    # extract the only transfer node at the moment
+    transfer_node = nodes[0]
+    assert transfer_node["type"] == "transfer"
+    # there are child nodes for each directory in the path
+    objects_dir_node = transfer_node["children"][0]
+    assert objects_dir_node["type"] == "directory"
+    # and finally a node for the file
+    file_node = objects_dir_node["children"][0]
+    # check that the file node properties match the record properties
+    _assert_file_node_properties_match_record(file_node, record)
+
+
+def test_appraisal_tab_node_draggability_logs():
+    """Test that the file nodes in the logs directory cannot be dragged"""
+    record = {
+        "relative_path": "transfer-directory/data/logs/arrange.log",
+        "fileuuid": "c58db760-3767-4284-a3a3-40b9cad61095",
+        "size": 1.0,
+        "tags": [],
+        "bulk_extractor_reports": [],
+        "modification_date": None,
+        "format": None,
+    }
+    mapping = {}
+    nodes = []
+    _es_results_to_appraisal_tab_format(record, mapping, nodes)
+    transfer_node = nodes[0]
+    data_dir_node = transfer_node["children"][0]
+    # hidden directories can't be dragged either
+    logs_dir_node = data_dir_node["children"][0]
+    assert logs_dir_node["not_draggable"]
+    file_node = logs_dir_node["children"][0]
+    assert file_node["not_draggable"]
+
+
+def test_appraisal_tab_node_draggability_metadata():
```

Comment:
Just a question about tests vs. visibility in the UI - the metadata dirs don't show up in the UI do they? (same for some of the "special" directories below?) 

(I can't see them, either way I do like this test being here for the future)

Reply:
You need to check the `Show metadata & logs directories?` checkbox before pressing the `Search transfer backlog` button in the Appraisal tab:

![Captura de pantalla de 2019-07-30 22-47-03](https://user-images.githubusercontent.com/560781/62184610-0ff9dc00-b31c-11e9-9c6c-6b7a7b2ae702.png)

Reply:
Oh! Well, I've learned something new! (two things if you include I should click more buttons in the UI :) Thanks @replaceafill.